src/v2/api/dtos/UpdateRoleDTO.ts

```diff
@@ -1,10 +1,15 @@
+import { ApiPropertyOptional } from '@nestjs/swagger';
 import { RoleName } from '@prisma/client';
 import { IsOptional } from 'class-validator';
 
 export class UpdateRoleDTO {
+  @ApiPropertyOptional({
+    enum: RoleName,
+  })
   @IsOptional()
     name?: RoleName;
 
+  @ApiPropertyOptional()
```

Comment:
Add IsNumber decorator and a validation option message
don't forget to add validation message to "update" endpoint to InvalidBodyException